### PR TITLE
Adds acceptance tests for component a11y audits

### PR DESCRIPTION
**Pull request by MelSumner**
### :pushpin: Summary

If merged, this PR would implement component a11y audits as acceptance tests based on the showcase/preview pages, and fixes preview pages as appropriate for accessibility.

Note: https://github.com/hashicorp/design-system/pull/1696 needs to be merged first. 

### :hammer_and_wrench: Detailed description

- moved side-nav and tabs test to separate JIRA Issues, will do those in separate PRs
- fixed the tag component padding (with guidance from design) so it passes conformance (2px right padding to 4px right padding)
 
### :link: External links

Jira ticket: [HDS-2554](https://hashicorp.atlassian.net/browse/HDS-2554)
Figma file: [if it applies]

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A11y tests have been run locally (`yarn test:a11y --filter="COMPONENT-NAME"`)
- n/a If documenting a new component, an acceptance test that includes the `a11yAudit` has been added
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2554]: https://hashicorp.atlassian.net/browse/HDS-2554?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/tests/acceptance/components/hds/alert-test.js
Comment (alex-ju) [inline packages/components/tests/acceptance/components/hds/alert-test.js:6]:
```suggestion
module('Acceptance | components | /hds/alert', function (hooks) {
```
for consistency with the accordion test

File: packages/components/tests/acceptance/components/hds/application-state-test.js
Comment (alex-ju) [inline packages/components/tests/acceptance/components/hds/application-state-test.js:6]:
```suggestion
module('Acceptance | components | hds/application state', function (hooks) {
```

File: packages/components/tests/acceptance/components/hds/alert-test.js
Comment (alex-ju) [inline packages/components/tests/acceptance/components/hds/alert-test.js:15]:
```suggestion
  test('Components/alert page passes automated a11y checks', async function (assert) {
```

File: packages/components/tests/acceptance/components/hds/application-state-test.js
Comment (alex-ju) [inline packages/components/tests/acceptance/components/hds/application-state-test.js:14]:
```suggestion
  test('Components/application-state page passes automated a11y checks', async function (assert) {
```

**Review by alex-ju — COMMENTED** [review]
This looks really good! Left a few nitpicking comments, but happy to approve if you'd like to do this in batches (instead to all in a single PR).

In terms of performance/time, running these locally I get the following values for the 3 modules: `6 tests completed in 3230 milliseconds`, so roughly 1s/component which is pretty good.

File: packages/components/tests/acceptance/components/hds/application-state-test.js
Comment (Dhaulagiri) [inline packages/components/tests/acceptance/components/hds/application-state-test.js:12]: [suggestion/$.02/off topic for the actual question we're trying to answer] acceptance tests get expensive in time, so I don't think this test provides a ton of value (we already test that this page renders as we expect in the percy snapshots, so the UR being correct or not isn't super valuable).

File: packages/components/tests/acceptance/components/hds/alert-test.js
Comment (MelSumner) [reply packages/components/tests/acceptance/components/hds/alert-test.js]: I went through and added this in a bulk-feature way.

File: packages/components/app/styles/components/tag.scss
Comment (alex-ju) [inline packages/components/app/styles/components/tag.scss:32]: This would need a changelog entry

Comment (Dhaulagiri) [reply packages/components/app/styles/components/tag.scss:32]: changes like this should be cherry-picked out into a separate PR

**Comment by MelSumner** [comment]
Moved side-nav (70 issues) and tabs (21 issues) to separate JIRA issues

File: packages/components/app/styles/components/tag.scss
Comment (MelSumner) [reply packages/components/app/styles/components/tag.scss:32]: 1. There is a changeset
2. I left it here because I felt it was directly related and a tiny change
3. I'll pull it out into a separate PR.

Comment (Dhaulagiri) [reply packages/components/app/styles/components/tag.scss]: From my perspective, this will make it easier to land that change more quickly/cleanly than tying it up in this PR which is more of an internally facing change that has a bunch of threads in it.

Comment (Dhaulagiri) [reply packages/components/app/styles/components/tag.scss]: I wouldn't have reverted the change here, but rather rebase this on top of the other PR/main once it is merged so that the fix is here but can get upstreamed faster.

File: packages/components/tests/acceptance/components/hds/link/inline-test.js
Comment (alex-ju) [inline packages/components/tests/acceptance/components/hds/link/inline-test.js:13]: It seems we agreed to remove these tests as we can detect if the page doesn't render from the regression tests.

File: packages/components/tests/acceptance/components/hds/link/standalone-test.js
Comment (alex-ju) [inline packages/components/tests/acceptance/components/hds/link/standalone-test.js:13]: Same here, we can drop this test before merging.

**Review by alex-ju — APPROVED** [review]
We have a test failing here (the one for Tags) that was addressed in #1696 so rebasing this branch to `main` should result in an overall pass. I spotted two instances where we still have 'the page renders' tests, otherwise is good to go in! 🎉

File: packages/components/tests/acceptance/components/hds/link/standalone-test.js
Comment (MelSumner) [reply packages/components/tests/acceptance/components/hds/link/standalone-test.js]: dropped 👍 thanks for catching that

File: packages/components/tests/acceptance/components/hds/link/inline-test.js
Comment (MelSumner) [reply packages/components/tests/acceptance/components/hds/link/inline-test.js]: dropped, thanks for catching that 👍

**Pull request merged**
